packages/dd-trace/src/platform/node/loader.js

```diff
@@ -57,8 +57,18 @@ class Loader {
 
         pkg = require(`${basedir}/package.json`)
 
-        const mainFile = path.posix.normalize(pkg.main || 'index.js')
-        if (!id.endsWith(`/node_modules/${instrumentation.name}/${mainFile}`)) continue
+        try {
+          const mainFile = path.posix.normalize(pkg.main || 'index.js')
+          const resolvedFile = require.resolve(path.posix.join(basedir, '../', instrumentation.name, mainFile))
```

Comment:
What happens if the module is nested in the `node_modules` of another module imported as a transitive dependency?